provider/storage_benchmark.py

```diff
@@ -0,0 +1,322 @@
+        else:
+            label = params.get('win_utils_label', 'WIN_UTILS')
+            drive_letter = utils_misc.get_winutils_vol(instance.session, label)
+            self.cmd = 'set nodosfilewarning=1 && set CYGWIN=nodosfilewarning'
+            self.iozone_path = drive_letter + r':\Iozone\iozone.exe'
+            self.setups = {'session_cmd': (self.cmd, 300)}
+
+
+class Iozone(StorageBenchmark):
+    def __init__(self, params, session):
+        super(Iozone, self).__init__(params['os_type'], session, 'iozone')
+        self.cfg = IozoneCfg(params['os_type'], params, self)
+        for method, args in self.cfg.setups.items():
+            getattr(self, method)(*args)
+
+    def run(self, cmd_options='-a', timeout=1800):
+        """
+        Run iozone test inside linux guest.
+
+        :param cmd_options: iozone command options, e.g: -azR -r 64k -n 1G -g
+                            1G -M -f /home/test
+        :type cmd_options: str
+        """
+        cmd = ' '.join((self.cfg.iozone_path, cmd_options))
+        return super(Iozone, self).run(cmd, timeout)
+
+
+class FioCfg(object):
+    def __init__(self, os_type, params, instance):
+        if os_type == 'linux':
+            self.download_url = 'git://github.com/axboe/fio.git'
+            self.download_path = os.path.join('/home', 'fio_repo')
+            self.fio_inst = os.path.join('/home', 'fio_inst')
+            self.link_src = '%s/bin/fio' % self.fio_inst
+            self.fio_path = '/usr/local/bin/fio'
+            self.setups = {'download_benchmark': (GIT_DOWNLOAD,
+                                                  self.download_url,
+                                                  self.download_path),
+                           'install': (self.download_path, self.fio_inst),
+                           'make_symlinks': (self.link_src, self.fio_path)}
+        else:
+            label = params.get('win_utils_label', 'WIN_UTILS')
+            utils_letter = utils_misc.get_winutils_vol(instance.session, label)
+            arch = params.get('vm_arch_name', 'x84_64')
+            self.fio_inst = {'x86_64': r'C:\Program Files (x86)\fio',
+                             'i686': r'C:\Program Files\fio'}
+            self.fio_msi = {'x86_64': r'%s:\fio-x64.msi' % utils_letter,
+                            'i686': r'%s:\fio-x86.msi' % utils_letter}
+            self.fio_path = r'"%s\fio\fio.exe"' % self.fio_inst[arch]
+            self.setups = {'install': (self.fio_msi[arch], self.fio_inst[arch], 300)}
+
+
+class Fio(StorageBenchmark):
+    def __init__(self, params, session):
+        super(Fio, self).__init__(params['os_type'], session, 'fio')
+        self.cfg = FioCfg(params['os_type'], params, self)
+        for method, args in self.cfg.setups.items():
+            getattr(self, method)(*args)
+
+    def run(self, cmd_options, timeout=1800):
+        """
+        Run fio test inside linux guest.
```

Comment:
same as iozone, should be typo for both windows and linux guests.

Reply:
thanks, have updated.